internal/dinosaur/pkg/gitops/service.go

```diff
@@ -26,12 +34,74 @@ func NewService(configProvider ConfigProvider) Service {
 }
 
 // GetCentral returns a Central instance with the given parameters.
-func (s *service) GetCentral(params CentralParams) (v1alpha1.Central, error) {
+func (s *service) GetCentral(params CentralParams) (string, error) {
 	cfg, err := s.configProvider.Get()
 	if err != nil {
-		return v1alpha1.Central{}, errors.Wrap(err, "failed to get GitOps configuration")
+		return "", errors.Wrap(err, "failed to get GitOps configuration")
+	}
+
+	result, err := readFromCache(params, cfg)
+	if err != nil {
+		return "", err
+	}
+	if result != "" {
+		return result, nil
+	}
+
+	centralCR, err := renderCentral(params, cfg)
+	if err != nil {
+		return "", err
+	}
+
+	centralYaml, err := yaml.Marshal(centralCR)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to marshal Central CR")
+	}
+
+	centralYAMLString := string(centralYaml)
+	err = addCache(params, cfg, centralYAMLString)
+	if err != nil {
+		return "", err
+	}
+	return centralYAMLString, nil
```

Comment:
I think you need to lock the cache key

```suggestion
        cacheKey, err := getCacheKey(params, cfg)
        if err != nil { return err }
        
        km.Lock(cacheKey)
        defer km.Unlock(cacheKey)

	result, err := readFromCache(cacheKey)
	if err != nil {
		return "", err
	}
	if result != "" {
		return result, nil
	}

	centralCR, err := renderCentral(params, cfg)
	if err != nil {
		return "", err
	}

	centralYaml, err := yaml.Marshal(centralCR)
	if err != nil {
		return "", errors.Wrap(err, "failed to marshal Central CR")
	}

	centralYAMLString := string(centralYaml)
	err = addCache(cacheKey, centralYAMLString)
	if err != nil {
		return "", err
	}
	return centralYAMLString, nil
```